src/main_thread/init/multi_thread_content_initializer.ts

```diff
@@ -1195,14 +1161,21 @@ export default class MultiThreadContentInitializer extends ContentInitializer {
         this._currentContentInfo.manifest,
         updates,
       );
-      sendMessage(this._settings.worker, {
-        type: MainThreadMessageType.DecipherabilityStatusUpdate,
-        contentId: this._currentContentInfo.contentId,
-        value: manUpdates.map((s) => ({
-          representationUniqueId: s.representation.uniqueId,
-          decipherable: s.representation.decipherable,
-        })),
-      });
+      if (
+        mayMediaElementFailOnUndecipherableData &&
+        manUpdates.some((e) => e.representation.decipherable !== true)
+      ) {
+        reloadMediaSource();
```

Comment:
Can this lead to reloading the media source if there is a decipherability change in a representation we were not playing?

Reply:
Ok, it's worth to add this little quirk while fixing the original bug